ego/cmd/bundle/main.go

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) Edgeless Systems GmbH.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//go:build linux
+// +build linux
+
+package main
+
+import (
+	"debug/elf"
+	"errors"
+	"fmt"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"syscall"
+
+	"ego/internal/launch"
+
+	"github.com/spf13/afero"
+)
+
+// Don't touch! Automatically injected at build-time.
+var (
+	version   = "0.0.0"
+	gitCommit = "0000000000000000000000000000000000000000"
+)
+
+// ErrExecutableDoesNotContainBundle is triggered when this executable is launched without containing a bundled enclave image as the ".ego.bundle" section.
+var ErrExecutableDoesNotContainBundle = errors.New("executable does not contain an enclave bundle")
+
+func main() {
+	fmt.Fprintf(os.Stderr, "EGo (bundled) v%v (%v)\n", version, gitCommit)
+
+	// Open the current execeutable
+	execPath, err := os.Executable()
+	if err != nil {
+		panic(err)
+	}
+
+	// Parse ourselves as an ELF file so we get access to ELF sections later
+	elfFile, err := elf.Open(execPath)
+	if err != nil {
+		panic(err)
+	}
+	defer elfFile.Close()
+
+	// Use the "host" environment and not a mocked one for unit tests
+	fs := afero.NewOsFs()
+	runner := launch.OsRunner{}
+
+	// Run the unpacking & launch the unpacked enclave.
+	// Take the error from the enclave as the exit code, unless we failed to execute the enclave before.
+	exitCode, err := run(fs, elfFile, runner)
+	if err != nil {
+		panic(err)
+	}
+	os.Exit(exitCode)
+}
+
+func run(fs afero.Fs, selfElfFile *elf.File, runner launch.Runner) (int, error) {
+	// Create a temporary directory for both runtime and enclave
+	tempEGoRootPath, err := afero.TempDir(fs, "", "ego-bundle")
+	if err != nil {
+		return 1, err
+	}
+	defer fs.RemoveAll(tempEGoRootPath)
+
+	// Register cleanup handler to clean-up on STRG+C
+	cleanupHandler(tempEGoRootPath)
+
+	// Unpack the runtime and enclave
+	if err := unpackBundledImage(fs, tempEGoRootPath, selfElfFile); errors.Is(err, ErrExecutableDoesNotContainBundle) {
+		fmt.Println("This file does not contain an enclave bundle and is not supposed to be run directly from EGo's installation directory.")
+		fmt.Println("It is only used for the output of the 'ego bundle' command.")
+		fmt.Println("Exiting...")
+		// Clear the error so that we do not panic.
+		return 3, nil
```

Comment:
We don't necessarily need this. The installed file should be treated as a data file and not be executable. I'd prefer to keep the code small and remove this.

Reply:
The error handling in unpackBundledImage should persist in my opinion, otherwise it is an unhandled nil when dealing with the section, causing a panic if the section is missing.

But if the error handling has to be there in any case, not sure if these additional Prints hurt too much?

Reply:
That's true I guess. I removed it, kept the custom error so the panic message should be descriptive enough for debugging, I believe.